src/Features/Core/Portable/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -69,28 +108,30 @@ internal abstract partial class AbstractInheritanceMarginService
             }
             else
             {
-                return await GetInheritanceMemberItemInProcessAsync(
-                    project, documentForGlobalImports, spanToSearch, symbolKeyAndLineNumbers, cancellationToken).ConfigureAwait(false);
+                return await GetSymbolItemsInProcessAsync(
+                    project, document, symbolKeyAndLineNumbers, frozenPartialSemantics, cancellationToken).ConfigureAwait(false);
             }
         }
 
-        private async ValueTask<ImmutableArray<InheritanceMarginItem>> GetInheritanceMemberItemInProcessAsync(
+        private static async ValueTask<ImmutableArray<InheritanceMarginItem>> GetSymbolItemsInProcessAsync(
             Project project,
-            Document? documentForGlobalImports,
-            TextSpan spanToSearch,
+            Document? document,
             ImmutableArray<(SymbolKey symbolKey, int lineNumber)> symbolKeyAndLineNumbers,
+            bool frozenPartialSemantics,
             CancellationToken cancellationToken)
         {
+            // If we're starting from a document, use it to go to a frozen partial version of it to lower the amount of
+            // work we need to do running source generators or producing skeleton references.
+            if (document != null && frozenPartialSemantics)
```

Comment:
I feel the 'frozenPartialSemantics' pattern is somewhat strange to me. (The flag seems always to be true)
Is this the pattern that Sam mentioned in the previous comment?